packages/jest-cli/src/cli/args.js

```diff
@@ -268,8 +267,8 @@ export const options = {
   lastCommit: {
     default: undefined,
     description:
-      'Will run all tests affected by file changes in the last ' +
-      'commit made.',
+      'When used together with `--onlyChanged`, it will run all tests ' +
+      'affected by file changes in the last commit made.',
```

Comment:
I noticed that both --lastCommit and --changedFilesWithAncestor are ignored if you don't also pass --onlyChanged or --watch. I thought about writing something in the `check()` function (line 15 in this file) explode if you pass an argument that will be ignored, but then I realised that someone might set --onlyChanged or --watch in their configs, or someone might add another arg that has the same effect, so I just decided to document it a bit harder.

I also decided against talking about `--watch` in the docs for `--lastCommit`, because it sounds like a pointless thing to do.

Reply:
>I noticed that both --lastCommit and --changedFilesWithAncestor are ignored if you don't also pass --onlyChanged or --watch.

That doesn't seem right. @cpojer WDYT?

Reply:
```
$ yarn jest --listTests | wc -l
     213
$ yarn jest --listTests --lastCommit | wc -l
     213
$ yarn jest --listTests --lastCommit --onlyChanged | wc -l
       4
```

@SimenB are you suggesting that: 
a) we should make `--lastCommit` and `--changedFilesWithAncestor` imply `--onlyChanged` (may cause previously valid configurations to test fewer things than they used to)
b) we should make `--lastCommit` and `--changedFilesWithAncestor` explode if it's not combined with `--onlyChanged` or `--watch` (may cause previously valid configurations to explode, and means that the next person to add an `--onlyChanged` synonym needs to update the check)

or are you just after a second reviewer?

Reply:
>or are you just after a second reviewer?

This 🙂

Reply:
Oh, interesting. I think the reason for that is because the default at FB is that onlyChanged is true and people must pass explicitly `--all` to run all tests. I guess we just never noticed the inconsistency. I think `--lastCommit --all` doesn't make any sense, so I'm in favor of the proposed change a).